arch/cortex-m/samv71-hal/src/xdmac/transfer.rs

```diff
@@ -0,0 +1,503 @@
+//! Module with transfer-related items.
+
+use crate::utils::{BoundedU16, BoundedU32};
+use samv71q21_pac::xdmac::{
+    self,
+    xdmac_chid::cc::{
+        CSIZESELECT_A, DAMSELECT_A, DIFSELECT_A, DSYNCSELECT_A, DWIDTHSELECT_A, MBSIZESELECT_A,
+        SAMSELECT_A, SIFSELECT_A, SWREQSELECT_A, TYPESELECT_A,
+    },
+};
+
+/// A single block of XDMAC transfer.
+///
+/// # Safety
+///
+/// **XDMAC transfers are inherently unsafe, as they operate directly on raw memory.**
+///
+/// In order to make sure that an XDMAC transfer will be performed safely, **you** must guarantee
+/// that the transfer configuration is valid, and will be valid during XDMAC transfer.
+/// This mostly means that **you** must make sure the source and destination addresses point to
+/// valid, aligned locations/buffers in memory, and the transfer is configured appropriately for the
+/// buffer(s) size.
+///
+/// **XDMAC will not perform any safety checks. [`TransferBlock`]'s constructor and setters will
+/// perform basic safety checks wherever possible, but it's not possible to fully validate the
+/// transfer configuration that way. Breaking Rust/XDMAC safety invariants will lead to an
+/// undefined behavior. Drive responsibly.**
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct TransferBlock {
+    /// Transfer's source
+    source: TransferLocation,
+
+    /// Transfer's destination
+    destination: TransferLocation,
+
+    /// Amount of data units in the microblock.
+    ///
+    /// Data unit size is configured in `data_width` field.
+    microblock_length: MicroblockLength,
+
+    /// Amount of microblocks in a transfer block.
+    block_length: BlockLength,
+
+    /// Transfer's type (memory-to-memory, peripheral-to-memory, memory-to-peripheral), and
+    /// peripheral ID in case of peripheral transfer.
+    transfer_type: TransferType,
+
+    /// Memory burst size.
+    memory_burst_size: MemoryBurstSize,
+
+    /// Channel's chunk size, amount of data transferred in a single chunk.
+    chunk_size: ChunkSize,
+
```

Comment:
```suggestion
    /// Transfer's destination
    destination: TransferLocation,
    /// Amount of data units in the microblock.
    ///
    /// Data unit size is configured in `data_width` field.
    microblock_length: MicroblockLength,
    /// Amount of microblocks in a transfer block.
    block_length: BlockLength,
    /// Transfer's type (memory-to-memory, peripheral-to-memory, memory-to-peripheral), and
    /// peripheral ID in case of peripheral transfer.
    transfer_type: TransferType,
    /// Memory burst size.
    memory_burst_size: MemoryBurstSize,
    /// Channel's chunk size, amount of data transferred in a single chunk.
    chunk_size: ChunkSize,
```